darts/models/transformer_model.py

```diff
@@ -0,0 +1,281 @@
+from .torch_forecasting_model import TorchForecastingModel
+
+logger = get_logger(__name__)
+
+
+# This implementation of positional encoding is taken from the PyTorch documentation:
+# https://pytorch.org/tutorials/beginner/transformer_tutorial.html
+class _PositionalEncoding(nn.Module):
+    def __init__(self, d_model, dropout=0.1, max_len=500):
+        """ An implementation of positional encoding as described in 'Attention is All you Need' by Vaswani et al. (2017)
+
+                Parameters
+                ----------
+                d_model
+                    the number of expected features in the transformer encoder/decoder inputs.
+                    Last dimension of the input
+                dropout
+                    Fraction of neurons affected by Dropout (default=0.1).
+                max_len
+                    The dimensionality of the computed positional encoding array. Only its first "input_size" elements will be
+                    considered in the output
+
+
+
+                Inputs
+                ------
+                x of shape `(batch_size, input_size, d_model)`
+                    Tensor containing the embedded time series
+
+                Outputs
+                -------
+                y of shape `(batch_size, input_size, d_model)`
+                    Tensor containing the embedded time series enhanced with positional encoding
+                """
+        super(_PositionalEncoding, self).__init__()
+        self.dropout = nn.Dropout(p=dropout)
+
+        pe = torch.zeros(max_len, d_model)
+        position = torch.arange(0, max_len, dtype=torch.float).unsqueeze(1)
+        div_term = torch.exp(torch.arange(0, d_model, 2).float() * (-math.log(10000.0) / d_model))
+        pe[:, 0::2] = torch.sin(position * div_term)
+        pe[:, 1::2] = torch.cos(position * div_term)
+        pe = pe.unsqueeze(0).transpose(0, 1)
+        self.register_buffer('pe', pe)
+
+    def forward(self, x):
+        x = x + self.pe[:x.size(0), :]
+        return self.dropout(x)
+
+
+class _TransformerModule(nn.Module):
+    def __init__(self,
+                 input_size: int,
+                 input_length: int,
+                 output_length: int,
+                 output_size: int,
+                 d_model: int,
+                 nhead: int,
+                 num_encoder_layers: int,
+                 num_decoder_layers: int,
+                 dim_feedforward: int,
+                 dropout: float,
+                 activation: str,
+                 custom_encoder: nn.Module = None,
+                 custom_decoder: nn.Module = None,
+                 ):
+        """ PyTorch module implementing a Transformer to be used in `TransformerModel`.
+
+        PyTorch module implementing a simple encoder-decoder transformer architecture.
+
+        Parameters
+        ----------
+        input_size
+            The dimensionality of the TimeSeries instances that will be fed to the fit function.
```

Comment:
We have this typo in several places, but in fact it should be "the fit and predict functions"